Proposal discussion fixes

traumschule opened the pull request:
- Merge `mainnet` for #3965
- #3236
- #3900
- #3970
- #3966 + improve `ProposalHistory`

![proposal-history](https://user-images.githubusercontent.com/31551045/207111247-7a5e24d2-6862-42da-b53e-a52a8d9f13a5.png)
=> https://pioneer-2-git-fork-traumschule-fix-reply-3967-joystream.vercel.app/#/proposals/preview/47

While loading:
![stake-loading](https://user-images.githubusercontent.com/31551045/207113195-7484f8a1-2ab6-4b04-82fb-a093740de9d9.png)

Commit:
Reverse order of proposal discussion posts (#3236)

Commit:
Proposal discussion mode tooltip text (#3900)

Commit:
Optimize hook order (#3970)

Commit:
Show proposal stake (#3966)

traumschule committed:
Merge branch 'dev' into fix/reply-3967

Commit:
ProposalHistory: Show block number (#3966)

oleksanderkorn reviewed (approved):
LGTM

traumschule commented:
[re](https://github.com/Joystream/pioneer/issues/3236#issuecomment-1346329412): revert 2743e59a3893255e76991046d07e757e1f1313ca?

traumschule commented:
[re](https://github.com/Joystream/pioneer/issues/3236#issuecomment-1346329412): revert 2743e59a3893255e76991046d07e757e1f1313ca?

thesan reviewed (changes requested):
@traumschule I'm confuse by the changes on this PR most of them are already implemented on `dev` (e.g `JOY` → `tJOY`) could you check what's happening ?

traumschule committed:
Merge branch 'dev' into fix/reply-3967

traumschule commented:
Fixed the conflict. @thesan as per description `dev` was merged to base on a former fix. Did you read [my other comment](https://github.com/Joystream/pioneer/pull/3969#issuecomment-1355646745)? Reverted it now according to the linked comment.

traumschule commented on packages/ui/src/proposals/components/ProposalDiscussions.tsx:
```diff
@@ -26,7 +26,7 @@ interface Props {
 
 const hints = {
-  open: 'This is an unmoderated discussioon, everyone can comment.',
+  open: 'This is an open discussion, everyone can comment.',
```
[according to](https://github.com/Joystream/pioneer/issues/3900#issuecomment-1406287531)
```suggestion
  open: 'Open discussion is open for every member to comment on, while closed would only be available for council members, white-listed members and creator of the proposal.',
```

traumschule committed:
fix tooltip text

thesan commented on packages/ui/src/proposals/components/ProposalHistory.tsx:
```diff
@@ -21,9 +21,8 @@ export const ProposalHistory = ({ proposal }: ProposalHistoryProps) => {
 
     return steps.map(({ status, inBlock }, index) => ({
-      title: status,
+      title: `${status} in block ${inBlock.number}`,
       type: index === steps.length - 1 ? 'active' : 'past',
```
About this part I replied in #3966.

But please reset this file (`git checkout dev -- packages/ui/src/proposals/components/ProposalHistory.tsx`) and implement the changes in another PR

traumschule replied:
sorry for the delay: bc91735

thesan commented on packages/ui/src/app/pages/Proposals/ProposalPreview.tsx:
```diff
@@ -195,6 +195,8 @@ export const ProposalPreview = () => {
             <RowGapBlock gap={16}>
               <Label>Proposer</Label>
               <MemberInfo member={proposal.proposer} />
+              <Label>Stake</Label>
+              {constants ? <TokenValue value={constants.requiredStake} /> : 'Loading...'}
             </RowGapBlock>
```
It's a really small issue but currently the spacing looks off ("STAKE" is closer to the proposer than the stake value):
```suggestion
            <RowGapBlock gap={16}>
              <Label>Proposer</Label>
              <MemberInfo member={proposal.proposer} />
            </RowGapBlock>

            <RowGapBlock gap={16}>
              <Label>Stake</Label>
              {constants ? <TokenValue value={constants.requiredStake} /> : 'Loading...'}
            </RowGapBlock>
```

thesan commented on packages/ui/src/app/pages/Proposals/ProposalPreview.tsx:
```diff
@@ -40,20 +41,19 @@ import { VoteRationaleModalCall } from '@/proposals/modals/VoteRationale/types'
+  const { id } = useParams<{ id: string }>()
   const { isLoading, proposal } = useProposal(id)
-  const { council } = useElectedCouncil()
   const constants = useProposalConstants(proposal?.details.type)
-  const loc = useLocation()
-  const voteId = new URLSearchParams(loc.search).get('showVote')
   const blocksToProposalExecution = useBlocksToProposalExecution(proposal, constants)
+  useRefetchQueries({ interval: MILLISECONDS_PER_BLOCK, include: ['getProposal', 'GetProposalVotes'] }, [proposal])
 
+  const { council } = useElectedCouncil()
+  const loc = useLocation()
+  const voteId = new URLSearchParams(loc.search).get('showVote')
   const votingRounds = useVotingRounds(proposal?.votes, proposal?.proposalStatusUpdates)
   const [currentVotingRound, setVotingRound] = useState(0)
-
   const votes = votingRounds[currentVotingRound] ?? votingRounds[0]
-  useRefetchQueries({ interval: MILLISECONDS_PER_BLOCK, include: ['getProposal', 'GetProposalVotes'] }, [proposal])
```
What are these changes for ?

traumschule replied:
375a07d was supposed to fix #3970

thesan reviewed (changes requested):
@traumschule please avoid grouping multiple issues in one PR. Smaller PR go through shorter review/changes cycles

traumschule committed:
spacing

Co-authored-by: Theophile Sandoz <theophile.sandoz@gmail.com>

Commit:
revert `ProposalHistory.tsx`

traumschule committed:
Merge branch 'dev' into fix/reply-3967

traumschule commented:
outdated
